src/main/scala/cromwell/engine/CallActor.scala

```diff
@@ -1,46 +1,79 @@
 package cromwell.engine
 
+import java.util.UUID
+import java.util.concurrent.Executors
+
 import akka.actor.{Actor, Props}
 import akka.event.{Logging, LoggingReceive}
 import cromwell.binding._
 import cromwell.binding.values.WdlValue
-import cromwell.engine
-import cromwell.engine.backend.Backend
+import cromwell.engine.backend.{TaskAbortedException, Backend}
 import cromwell.engine.workflow.WorkflowActor
 import cromwell.engine.workflow.WorkflowActor.CallFailed
 
+import scala.collection.mutable
+import scala.concurrent.{ExecutionContext, Future}
 import scala.language.postfixOps
-import scala.util.{Failure, Success}
-
+import scala.util.{Try, Failure, Success}
 
 object CallActor {
+
   sealed trait CallActorMessage
-  case class Start(inputs: CallInputs) extends CallActorMessage
-  case class ExecutionFinished(outputs: CallOutputs) extends CallActorMessage
-  case class ExecutionFailed(regret: Throwable) extends CallActorMessage
+  case class Start(call: Call, locallyQualifiedInputs: Map[String, WdlValue], backend: Backend, workflowDescriptor: WorkflowDescriptor) extends CallActorMessage
+  case class RegisterCallAbortFunction(executionReference: CallReference, abortFunction: AbortFunction) extends CallActorMessage
+  case class AbortCall(workflowId: WorkflowID) extends CallActorMessage
+  case class ExecutionFinished(callReference: CallReference, call: Call, outputs: Try[CallOutputs]) extends CallActorMessage
 
-  def props(call: Call, locallyQualifiedInputs: Map[String, WdlValue], backend: Backend, workflowDescriptor: WorkflowDescriptor): Props =
-    Props(new CallActor(call, locallyQualifiedInputs, backend, workflowDescriptor))
+  def props(): Props = Props(new CallActor())
 }
 
 /** Actor to manage the execution of a single call. */
-class CallActor(call: Call, locallyQualifiedInputs: Map[String, WdlValue], backend: Backend, workflowDescriptor: WorkflowDescriptor) extends Actor with CromwellActor {
+class CallActor() extends Actor with CromwellActor {
+
+  import CallActor._
 
   type CallOutputs = Map[String, WdlValue]
 
+  val callActorId: UUID = UUID.randomUUID()
```

Comment:
I think it's clear here that you're assigning a type of `UUID`, no need to annotate